### PR TITLE
pAI improvements and fixes

### LINKED ISSUE #2772: pAI Bugs (Character Setup)
**Checklist before you submit an issue! Feel free to partially or fully delete this in your eventual report.**
* **Do a quick key word search of the git for duplicate reports. If you find any, simply post a reply onto that issue instead!** N/A
* **Please be specific in your description of the issue. Explain the desired result (what should be happening) and the actual result (what is happening). Simply stating that [something happens] is not very helpful at describing what's wrong.** In character setup, you can set your pAI name, description, roles, and OOC comments. However, when I was selected for a pAI these were not auto filled, I had to copy and paste them in. 
* **Have you tried reproducing the issue? If yes, steps to reproduce it would help immensely!** This has happened over two rounds. 
* **All additional details help. Such as round ID, an admin you talked to if the issue required immediate fixing, etcetera, etcetera.** Admin: Alberyk   Round ID: bO4-c9ZW

### PR DESCRIPTION
**Pull request by Karolis2011**
rebranded and reduced #6814 
- [x] Fix pAI preferences not working (fixes #2772) 
- [x] Ports pAI submission window to VueUi

### CODE REVIEW

**Comment by Karolis2011** [comment]
!review

File: code/modules/mob/abstract/observer/observer.dm [hunk 1]
```diff
@@ -926,4 +926,24 @@ mob/abstract/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)
...
 		return
 
-	SSghostroles.vui_interact(src)
\ No newline at end of file
+	SSghostroles.vui_interact(src)
+
+/mob/abstract/observer/verb/submitpai() //Moves the ghost instead of just changing the ghosts's eye -Nodrak
+	set category = "Ghost"
+	set name = "Submit pAI personality"
+	set desc = "Submits pAI personality to pAI candidite pool"
```
Comment (NonQueueingMatt) [inline code/modules/mob/abstract/observer/observer.dm]:
```suggestion
	set desc = "Submits you pAI personality to the pAI candidate pool."
```

File: code/modules/mob/abstract/observer/observer.dm [hunk 2]
```diff
@@ -926,4 +926,24 @@ mob/abstract/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)
...
+
+	if(!src.MayRespawn(TRUE))
+		return
+	if(jobban_isbanned(src, "pAI"))
+		to_chat(src, "You are job banned for pAI position.")
```
Comment (NonQueueingMatt) [inline code/modules/mob/abstract/observer/observer.dm]:
```suggestion
		to_chat(src, "You are job banned from the pAI position.")
```

File: code/modules/mob/abstract/observer/observer.dm [hunk 3]
```diff
@@ -926,4 +926,24 @@ mob/abstract/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)
...
+		return
+	SSpai.recruitWindow(src)
+
+/mob/abstract/observer/verb/revokepai() //Moves the ghost instead of just changing the ghosts's eye -Nodrak
+	set category = "Ghost"
+	set name = "Revoke pAI personality"
+	set desc = "Removes you from pAI candidite pool"
```
Comment (NonQueueingMatt) [inline code/modules/mob/abstract/observer/observer.dm]:
```suggestion
	set desc = "Removes you from the pAI candidite pool."
```

File: code/modules/mob/abstract/observer/observer.dm [hunk 4]
```diff
@@ -926,4 +926,24 @@ mob/abstract/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)
...
+	set desc = "Submits pAI personality to pAI candidite pool"
+
+	if(!src.MayRespawn(TRUE))
+		return
+	if(jobban_isbanned(src, "pAI"))
+		to_chat(src, "You are job banned for pAI position.")
+		return
+	SSpai.recruitWindow(src)
+
+/mob/abstract/observer/verb/revokepai() //Moves the ghost instead of just changing the ghosts's eye -Nodrak
```
Comment (NonQueueingMatt) [inline code/modules/mob/abstract/observer/observer.dm]:
```suggestion
/mob/abstract/observer/verb/revokepai()
```

File: code/modules/mob/abstract/observer/observer.dm [hunk 5]
```diff
@@ -926,4 +926,24 @@ mob/abstract/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)
 		to_chat(usr, "<span class='danger'>The round hasn't started yet!</span>")
 		return
 
-	SSghostroles.vui_interact(src)
\ No newline at end of file
+	SSghostroles.vui_interact(src)
+
+/mob/abstract/observer/verb/submitpai() //Moves the ghost instead of just changing the ghosts's eye -Nodrak
```
Comment (NonQueueingMatt) [inline code/modules/mob/abstract/observer/observer.dm]:
```suggestion
/mob/abstract/observer/verb/submitpai()
```

File: code/modules/mob/abstract/observer/observer.dm [hunk 6]
```diff
@@ -926,4 +926,24 @@ mob/abstract/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)
...
+		return
+	SSpai.recruitWindow(src)
+
+/mob/abstract/observer/verb/revokepai() //Moves the ghost instead of just changing the ghosts's eye -Nodrak
+	set category = "Ghost"
+	set name = "Revoke pAI personality"
+	set desc = "Removes you from pAI candidite pool"
+
+	if(SSpai.revokeCandidancy(src))
+		to_chat(src, "You been removed from pAI condidite pool.")
```
Comment (NonQueueingMatt) [inline code/modules/mob/abstract/observer/observer.dm]:
```suggestion
		to_chat(src, "You been removed from the pAI candidate pool.")
```

File: html/changelogs/fsdfsf.yml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+author: Karolis2011
+delete-after: True
+changes: 
+  - rscadd: "Added verbs to ghost meniu to submit pAI personality and revoke it."
```
Comment (NonQueueingMatt) [inline html/changelogs/fsdfsf.yml]:
```suggestion
  - rscadd: "Added verbs to ghost menu to submit pAI personality and revoke it."
```

File: html/changelogs/fsdfsf.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+  - rscadd: "Added new VueUI for pAI personality submission."
+  - bugfix: "Fixed pAI prefrences."
```
Comment (NonQueueingMatt) [inline html/changelogs/fsdfsf.yml]:
```suggestion
  - bugfix: "Fixed pAI preferences."
```

File: code/controllers/subsystems/pai.dm [hunk 9]
```diff
@@ -136,116 +124,28 @@
...
+
+	var/datum/vueui/ui = SSvueui.get_open_ui(M, src)
+	if(!ui)
+		ui = new(M, src, "misc-pai-recruit", 580, 590, "pAI Personality Configuration", list(
+			"name" = candidate.name || "",
+			"description" = candidate.description || "",
+			"role" = candidate.role || "",
+			"comments" = candidate.comments || ""
+		), nstate = interactive_state)
+		ui.metadata = list("candidite" = candidate)
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/pai.dm]: `Candidate` is spelled wrong throughout the entirety of this.

File: code/controllers/subsystems/pai.dm [hunk 10]
```diff
@@ -136,116 +124,28 @@
...
+			candidate.comments = sanitize(pai["comments"])
+
+	var/datum/vueui/ui = SSvueui.get_open_ui(M, src)
+	if(!ui)
+		ui = new(M, src, "misc-pai-recruit", 580, 590, "pAI Personality Configuration", list(
+			"name" = candidate.name || "",
+			"description" = candidate.description || "",
+			"role" = candidate.role || "",
+			"comments" = candidate.comments || ""
+		), nstate = interactive_state)
```
Comment (NonQueueingMatt) [inline code/controllers/subsystems/pai.dm]: Could push this parenthesis up for better readability? Not really a blocking concern though, up to you.

File: code/modules/mob/abstract/observer/observer.dm [hunk 11]
```diff
@@ -926,4 +926,24 @@ mob/abstract/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)
...
-	SSghostroles.vui_interact(src)
\ No newline at end of file
+	SSghostroles.vui_interact(src)
+
+/mob/abstract/observer/verb/submitpai() //Moves the ghost instead of just changing the ghosts's eye -Nodrak
+	set category = "Ghost"
+	set name = "Submit pAI personality"
+	set desc = "Submits pAI personality to pAI candidite pool"
+
+	if(!src.MayRespawn(TRUE))
```
Comment (NonQueueingMatt) [inline code/modules/mob/abstract/observer/observer.dm]: Maybe set a `MINISYNTH` or `ANIMAL` respawn type here? So people don't have to wait the full time to become a pAI.

File: code/modules/mob/abstract/observer/observer.dm [hunk 12]
```diff
@@ -926,4 +926,22 @@ mob/abstract/observer/MayRespawn(var/feedback = 0, var/respawn_type = null)
...
+		return
+	SSpai.recruitWindow(src)
+
+/mob/abstract/observer/verb/revokepai()
+	set category = "Ghost"
+	set name = "Revoke pAI personality"
+	set desc = "Removes you from the pAI candidite pool."
+
+	if(SSpai.revokeCandidancy(src))
+		to_chat(src, "You been removed from the pAI candidate pool.")
```
Comment (NonQueueingMatt) [inline code/modules/mob/abstract/observer/observer.dm]:
```suggestion
		to_chat(src, "You have been removed from the pAI candidate pool.")
```

**Pull request merged**